### ISSUE TITLE
Content change not detected

### ISSUE THREAD

**Issue by jfranknichols** [CineDev/ParagraphTextKit#1]
If I copy and paste a word within the paragraph, and the paste is over a different word of the same length the change is not detected, or at least not reported as an edit to the delegate. This results in lost onscreen edits, since the backing store is not updated.

**Comment by CineDev** [comment]
Please, check the case now on version 1.1.1

**Comment by jfranknichols** [comment]
Okay, the problem still exists, here is how I duplicate it:

I am working with a master/detail view arrangement. I click on the master list which sends text into a NSTextView (where I added your ParagraphTextKit) which then eventually updates the NSView text field as:

```
    func updateNSView(_ view: StringEditorView, context: Context) {
        view.text = text
        view.selectedRanges = context.coordinator.selectedRanges
    }
```

I currently have my data model connected to only send its data to the detail view and not to update from any edits, just to reduce confusion. I added a simple array (editorArray) to track your updates and output. Here is my simple model and my modified version of your delegate methods, where I have added debug print statements:

```
   // ParagraphTextKit Delegte Methods
    //
    var editorModel = [String]()
    
    func paragraphCount() -> Int {
        return editorModel.count
    }
    
    func textStorage(_ textStorage: ParagraphTextStorage, didChangeParagraphs changes: [ParagraphTextStorage.ParagraphChange]) {
        for change in changes {
            switch change {
            case .insertedParagraph(index: let index, descriptor: let paragraphDescriptor):
                if editorModel.count > 0 {
                    print("before insert - editorModel[\(index)]='\(editorModel[index])'")
                } else {
                    print("before insert - editModel is empty")
                }
                print("insert at index[\(index)]='\(paragraphDescriptor.text)'")
                editorModel.insert(paragraphDescriptor.text, at: index)
                print("after insert - editorModel[\(index)]='\(editorModel[index])'")
                print("\n")
                
            case .removedParagraph(index: let index):
                editorModel.remove(at: index)
                print("remove:")
                
            case .editedParagraph(index: let index, descriptor: let paragraphDescriptor):
                if editorModel.count > 0 {
                    print("before edit - editorModel[\(index)]='\(editorModel[index])'")
                } else {
                    print("before edit - editModel is empty")
                }
                print("edit at index [\(index)]='\(paragraphDescriptor.text)'")
                editorModel[index] = paragraphDescriptor.text
                print("after insert - editorModel[\(index)]='\(editorModel[index])'")
                print("\n")
            }
        }
    }

```

And here is the output of the prints - the first is when I click the first master line "A Novel Template".  I see you insert an empty paragraph first, and then edit the empty paragraph to add the text. These two events happen twice below. I expect it is from my master view sending the text twice, so I am not worried about that. (my bug, I will fix later)

```
2020-11-19 16:32:42.533552-0500 Hermes[15293:1020809] Metal API Validation Enabled
before insert - editModel is empty
insert at index[0]=''
after insert - editorModel[0]=''


before edit - editorModel[0]=''
edit at index [0]='A Novel Template: default placeholder - please change'
after insert - editorModel[0]='A Novel Template: default placeholder - please change'


before insert - editModel is empty
insert at index[0]=''
after insert - editorModel[0]=''


before edit - editorModel[0]=''
edit at index [0]='A Novel Template: default placeholder - please change'
after insert - editorModel[0]='A Novel Template: default placeholder - please change'
```
Next, in the detail view, I select and copy the word "change" and then select and paste it over the word "please" - these words are the same length (If I paste a word that is a different length the change is reported via the delegate). 

No output from the paste/edit occurs.

I then select a different master row - "Nightlfall at Limelist" to see if the change was pending and sending new text to the detail view would result in flushing it. It doesn't. Only the new row out is printed as expected. So the "edit" of the work "Please" was lost.

```
before insert - editModel is empty
insert at index[0]=''
after insert - editorModel[0]=''


before edit - editorModel[0]=''
edit at index [0]='Nightfall at Limelight: default placeholder - please change'
after insert - editorModel[0]='Nightfall at Limelight: default placeholder - please change'


before insert - editModel is empty
insert at index[0]=''
after insert - editorModel[0]=''


before edit - editorModel[0]=''
edit at index [0]='Nightfall at Limelight: default placeholder - please change'
after insert - editorModel[0]='Nightfall at Limelight: default placeholder - please change'
```

Again, I apologize for the duplicate updates sent to the detail view. I am running in SwiftUI with the detail view wrapped in a RepresentableView wrapper, and sometimes SwiftUI will rebuild displays multiple times - I am ignoring them for now, since the output from your code makes sense.

**Comment by CineDev** [comment]
Sorry, somehow the fix hasn't appear in 1.1.1. Just update to 1.1.2 and it will do the job!

**Comment by jfranknichols** [comment]
Thank you, that fixed the problem. I appreciate your help.
